crates/services/producer/src/block_producer/tests.rs

```diff
@@ -543,6 +543,110 @@ mod produce_and_execute_block_txpool {
     }
 }
 
+// Tests for the `dry_run` method.
+mod dry_run {
+    use super::*;
+
+    #[tokio::test]
+    async fn dry_run__executes_with_given_timestamp() {
+        // Given
+        let simulated_block_time = Tai64::from_unix(1337);
+        let executor = MockExecutorWithCapture::default();
+        let ctx = TestContext::default_from_executor(executor.clone());
+
+        // When
+        let _err = ctx
+            .producer()
+            .dry_run(vec![], None, Some(simulated_block_time), None, None)
+            .await
+            .expect("dry run failed");
```

Comment:
Since I changed the executor to return `Ok`, this is a normal `expect` so it should only panic if the dry run has failed. Though thinking about this we don't care about whether this returns an error or not here so I'll change this to just ignore the results instead.

Reply:
Changed in https://github.com/FuelLabs/fuel-core/pull/2206/commits/24d509552fadf33412bb3bc488ebac738c839f96